File: Math, Logical Implememt/Notation/acmicpc.net/1373/Main.py
# import sys
# input = sys.stdin.readline
file = open("./01 ing/acmicpc.net/sample.txt", "r")
input = file.readline

num = list(map(int, input()))
n = len(num)
if(n % 3 == 1):
    print(num[0], end="")
elif(n % 3 == 2):
    tmp = 2 + num[1]
    print(tmp, end="")

for i in range(n % 3, n, 3):
    tmp = num[i] * 4 + num[i+1]*2 + num[i+2]
    print(tmp, end="")

File: Math, Logical Implememt/Notation/acmicpc.net/1373/Main.js
"use strict";

const inputFilePath = "/sample.txt";
const input = require("fs")
  .readFileSync(__dirname + inputFilePath)
  .toString()
  .trim()
  .split("\n");
// const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");
const num = input[0].split("").map((x) => Number(x));
num.reverse();
const ans = [0];
let cnt = 0;
let idx = 0;
for (let i = 0; i < num.length; i++) {
  if (cnt === 3) {
    cnt = 0;
    idx++;
    ans[idx] = 0;
  }
  if (num[i] === 1) ans[idx] = ans[idx] + Math.pow(2, i % 3);
  cnt++;
}
ans.reverse();
console.log(ans.join(""));
